46.md

```diff
@@ -1,98 +1,163 @@
+| `sign_event`             | `[<json_stringified_event_to_sign>]`              | `json_stringified(<signed_event>)`                                     |
+| `ping`                   | `[]`                                              | "pong"                                                                 |
+| `get_relays`             | `[]`                                              | `json_stringified({<relay_url>: {read: <boolean>, write: <boolean>}})` |
+| `get_public_key`         | `[]`                                              | `<hex-pubkey>`                                                         |
+| `nip04_encrypt`          | `[<third_party_pubkey>, <plaintext_to_encrypt>]`  | `<nip04_ciphertext>`                                                   |
+| `nip04_decrypt`          | `[<third_party_pubkey>, <ciphertext_to_decrypt>]` | `<plaintext>`                                                          |
+| `nip44_conversation_key` | `[<third_party_pubkey>]`                          | `<nip44_conversation_key>`                                             |
+| `nip44_encrypt`          | `[<third_party_pubkey>, <plaintext_to_encrypt>]`  | `<nip44_ciphertext>`                                                   |
+| `nip44_decrypt`          | `[<third_party_pubkey>, <ciphertext_to_decrypt>]` | `<plaintext>`                                                          |
+
+## Response Events `kind:24133`
+
+```json
+{
+    "id": <id>,
+    "kind": 24133,
+    "pubkey": <remote_signer_pubkey>,
+    "content": <nip04(<response>)>,
+    "tags": [["p", <local_keypair_pubkey>]],
+    "created_at": <unix timestamp in seconds>,
+}
+```
+
+The `content` field is a JSON-RPC-like message that is [NIP-04](https://github.com/nostr-protocol/nips/blob/master/04.md) encrypted and has the following structure:
+
+```json
+{
+    "id": <request_id>,
+    "result": <results_string>,
+    "error": <error_string>
+}
+```
+
+- `id` is the request ID that this response is for.
+- `results` is a string of the result of the call (this can be either a string or a JSON stringified object)
+- `error` is an error in string form.
+
+### Auth Challenges
+
+An Auth Challenge is a response that a remote signer can send back when it needs the user to authenticate via other means. This is currently used in the OAuth-like flow enabled by signers like [Nsecbunker](https://github.com/kind-0/nsecbunkerd/). The response `content` object will take the following form:
+
+```json
+{
+    "id": <request_id>,
+    "result": "auth_url",
+    "error": <URL_to_display_to_end_user>
+}
+```
+
+Clients should display (in a popup or new tab) the URL from the `error` field and then subscribe/listen for another response from the remote signer (reusing the same request ID). This event will be sent once the user authenticates in the other window (or will never arrive if the user doesn't authenticate).
```

Comment:
Clients can add an extra `redirect_uri` to get the bunker to redirect the user to that URI after account creation. This is useful in popup/new-tab-blocking contexts where all the client is able to do is redirect the user to a different URL.

Reply:
I'll add a note.